PageObjects/CartPage.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Microsoft.Playwright;
+using NUnit.Framework;
+using TechTalk.SpecFlow;
+
+namespace HW18_SpecFlow.PageObjects
+{
+    [Binding]
+    internal class CartPage
+    {
+        private readonly IPage page;
+
+        public CartPage(IPage page)
+        {
+            this.page = page;
+        }
+
+        public async Task VerifyPageUrl(string testPageUrl)
+        {
+            try
+            {
+                await page.WaitForURLAsync(testPageUrl);
+            }
+            catch (PlaywrightException e)
```

Comment:
you have "Then" step with _CartPage.VerifyPageUrl
and implementation of this step is try/catch which just gulps exception and makes this step to be always green. Moreover, for some specific error message you write it into console and for all other cases just do nothing. So whatever exception happens (if it does not contain "crash") this step just ignores everything. 
There is no need in try/catch constructions in tests at all. If something goes wrong, test must fail